test/test_pub.dart

```diff
@@ -81,8 +81,10 @@ String packageSpecLine(String packageName) => File(d.path(packagesFilePath))
 class RunCommand {
   static final get = RunCommand(
       'get', RegExp(r'Got dependencies!|Changed \d+ dependenc(y|ies)!'));
-  static final upgrade = RunCommand('upgrade',
-      RegExp(r'(No dependencies changed\.|Changed \d+ dependenc(y|ies)!)$'));
+  static final upgrade = RunCommand('upgrade', RegExp(r'''
+(No dependencies changed\.|Changed \d+ dependenc(y|ies)!)($|
+\d packages? (has|have) newer versions incompatible with dependency constraints.
```

Comment:
I guess we never have a test with more than 9 newer packages, but this should be:
```
\d+ packages?
```

Reply:
Done!